aviary/interface/methods_for_level2.py

```diff
@@ -487,6 +487,57 @@ def check_and_preprocess_inputs(self):
         # Fill in anything missing in the options with computed defaults.
         preprocess_crewpayload(self.aviary_inputs)
 
+        if self.mission_method is HEIGHT_ENERGY:
+            self.phase_separator()
+
+        # Target_distance verification for all phases
+        # Checks to make sure target_distance is positive,
+        for idx, phase_name in enumerate(self.phase_info):
+            if 'user_options' in self.phase_info[phase_name]:
+                if 'target_distance' in self.phase_info[phase_name]["user_options"]:
+                    target_distance = self.phase_info[phase_name]["user_options"]["target_distance"]
+                    if target_distance[0] <= 0:
+                        raise ValueError(
+                            f"Invalid target_distance in [{phase_name}].[user_options]. "
+                            f"Current (value: {target_distance[0]}), (units: {target_distance[1]}) <= 0")
+
+        # Target_time verification for all phases
+        # Checks to make sure target_time is positive,
+        # duration_bounds and initial_guesses for time have not been set,
+        # Then sets duration_bounds, initial_guesses, and fixed_duration
+        for idx, phase_name in enumerate(self.phase_info):
+            if 'user_options' in self.phase_info[phase_name]:
+                if 'target_time' in self.phase_info[phase_name]["user_options"]:
+                    target_time = self.phase_info[phase_name]["user_options"]["target_time"]
+                    if target_time[0] <= 0:
+                        raise ValueError(
+                            f"Invalid target_time in phase_info[{phase_name}][user_options]."
+                            f"Current (value: {target_time[0]}), (units: {target_time[1]}) <= 0")
+                    if 'duration_bounds' in self.phase_info[phase_name]["user_options"]:
+                        # raise ValueError(
+                        print(
+                            f"When specifying target_time, duration_bounds for time should be removed. "
```

Comment:
I think the examples have both set?

Reply:
The examples currently trigger these errors because `check_phase_info()` still needs a lot of work, more than I can accomplish this job. That's why the `ValueError()` is commented out, otherwise the examples would crash.